Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -191,21 +191,22 @@ private EmbedBuilder BuildPollEmbed(PromotionCampaignSummary campaign, int Appro
                 return error == null ? embed : embed.AddField("⚠️ Vote Error: ⚠️", error);
         }
 
-        private async Task<ITextChannel> GetPollChannel(IGuild guild)
+        private async Task<ITextChannel> GetPollChannelAsync(IGuild guild)
         {
             var getPollChannel = await _designatedChannelService
                 .GetDesignatedChannelsAsync(guild, DesignatedChannelType.PromotionInterface);
             return getPollChannel.First() as ITextChannel;
         }
 
-        private void SetDialogCache(long campaign, ulong message, CachedPromoDialog dialog)
+        private void SetDialogCache(CachedPromoDialog dialog)
         {
-            _memoryCache.Set(campaign, dialog);
-            _memoryCache.Set(message, dialog);
+            _memoryCache.Set(GetKey(dialog.CampaignId), dialog);
+            _memoryCache.Set(dialog.MessageId, dialog);
```

Comment:
This still is not globally unique.

Reply:
how so?

Reply:
MessageId can be used in multiple contexts other than just promotions. For instance, what if we wanted to do this same caching for Starboard messages

Reply:
would it not be a diff id? and also not a promoDialog anon object

Reply:
Why would the ID for the same message in two different parts of the application not be the same?

> and also not a promoDialog anon object

Exactly, so we'd get a runtime exception when it tried to cast an object that isn't a promotion dialog to a promotion dialog... which is bad

Reply:
To clarify, we're talking about this line:
`_memoryCache.Set(dialog.MessageId, dialog);`
which is just using a plain MessageId as the key

Reply:
I.E. you create a `GetKey()` method for one of the two things you're caching, but not both.